backend/LexBoxApi/GraphQL/CustomTypes/ProjectMembersVisibilityMiddleware.cs

```diff
@@ -0,0 +1,25 @@
+using HotChocolate.Resolvers;
+using LexBoxApi.Auth;
+using LexCore.Entities;
+using LexCore.Exceptions;
+using LexCore.ServiceInterfaces;
+
+namespace LexBoxApi.GraphQL.CustomTypes;
+
+public class ProjectMembersVisibilityMiddleware(FieldDelegate next)
+{
+    public async Task InvokeAsync(IMiddlewareContext context, IPermissionService permissionService, LoggedInContext loggedInContext)
+    {
+        await next(context);
+        if (context.Result is IEnumerable<ProjectUsers> projectUsers)
+        {
+            var contextProject = context.Parent<Project>();
+            var projId = contextProject?.Id ?? throw new RequiredException("Must include project ID in query if querying users");
+            if (!await permissionService.CanViewProjectMembers(projId))
+            {
+                // Confidential project, and user doesn't have permission to see its users, so only show the current user's membership
+                context.Result = projectUsers.Where(pu => pu.User?.Id == loggedInContext.MaybeUser?.Id);
```

Comment:
this could possibly do weird stuff due to lazy evaluation and variable capturing. Since we're filtering with Where then an enumerable is getting assigned to the result but not evaluated until _later_ maybe later when the logged in context is out of scope, I'm not really sure? We should just call `ToList` and assign that to the result to avoid any potential issues.

Reply:
Yes, I was a bit unsure about this, but forgot. Thanks for noticing!

Reply:
c7015d17e97a472f49632d843880acbb4ec1c42d `ToList()`